### ISSUE TITLE
GripQL 'jump' function

### ISSUE THREAD

**Issue by kellrott** [bmeg/grip#155]
Add method to allow for jumping back to previously marked vertices.
Example Query:
```
O.query().V().mark("a").out("controller").mark("b").jump("a").out("controlled").mark("c").render(["$a", "$b", "$c"])
```

**Comment by kellrott** [comment]
In gremlin, `select` is used for building an output array and for jumping to a previous step ( http://tinkerpop.apache.org/docs/current/reference/#select-step ). Currently, the `select` implementation in grip only does the row output formatting (which is redundant with the `render` method) and doesn't implement the jump like function. 
This needs to be cleared up and fixed.

**Comment by adamstruck** [comment]
fixed in #166
